### ISSUE TITLE
failed to install Docker in RHEL and SUSE for RKE-provisioned cluster 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#24529]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- run rancher:master-head f3fbe1449 
- add 2 node templates for AWS nodes:
  - AMI `RHEL-7.7_HVM_GA-20190723-x86_64-1-Hourly2-GP2`; SSH user: ec2-user
  - AMI `suse-sles-12-sp2-v20161214-ecs-hvm-ssd-x86_64`:  SSH user: ec2-user
- create 2 clusters using the each of the above node templates 

**Result:**
- Rancher fails to install Docker in both clusters 
- For the `RHEL` cluster, Rancher logs and the UI show error message, and Rancher keeps deleting and creating new instances in AWS 
```
2019/12/11 17:04:06 [ERROR] NodeController c-9x85j/m-8r5l6 [node-controller] failed with : Error creating machine: Error running provisioning: error installing docker:
```
<img width="1457" alt="Screen Shot 2019-12-11 at 10 04 20 AM" src="https://user-images.githubusercontent.com/6218999/70643183-c9878880-1bfd-11ea-8572-21eae6497952.png">

- For the `SUSE` cluster, the cluster fails at the health check stage for worker nodes. The error message shows an old version of Docker, not the one we set in the node template, is used 

```
2019/12/11 17:21:16 [ERROR] ClusterController c-82b7k [cluster-provisioner-controller] failed with : [workerPlane] Failed to bring up Worker Plane: [Failed to verify healthcheck: Failed to check http://localhost:10248/healthz for service [kubelet] on host [3.135.206.199]: Get http://localhost:10248/healthz: dial tcp [::1]:10248: connect: connection refused, log: F1211 17:21:03.371518   18279 server.go:271] failed to run Kubelet: failed to create kubelet: docker API version is older than 1.26.0]
```
<img width="1476" alt="Screen Shot 2019-12-11 at 10 21 43 AM" src="https://user-images.githubusercontent.com/6218999/70644454-46b3fd00-1c00-11ea-80d3-eb12f629c25f.png">

More Info:
- Both images are from the community
- the SSH User for both is `ec2-user`
- the `Docker Install URL` in the node template is the default value which is `https://releases.rancher.com/install-docker/19.03.sh`
- Rancher can install Docker in `Ubuntu` and `RancherOS` and provision clusters successfully

**Comment by superseb** [comment]
For RHEL, the repository that is needed for dependencies for the `docker-ce` package live in the `rhel-7-server-rhui-extras-rpms` repository. We need to enable this in our `install-docker` scripts. As soon as 19.03.5 PR is merged, I can submit a new PR for all 19.03 versions to include enabling tis repository.

For SLES, the AMI mentioned comes with Docker pre-installed (as were two others that I tried), and because the AMI is that old, you get a very old Docker. The install script does not support SuSE and will just error out, and the Docker version already installed will be used which is 1.12 and is not supported.

**Comment by jiaqiluo** [comment]
the fix is validated in v2.3.4-rc5 and master-head 189e4337c with the default `docker install script`: `https://releases.rancher.com/install-docker/19.03.sh`

For RHEL, the AMI is the same as the issue provides: `RHEL-7.7_HVM_GA-20190723-x86_64-1-Hourly2-GP2`, Docker 19.3.5 is installed successfully, and the cluster is up and active with the k8s version 1.16.4

For SLES, since the install script does not support SUSE, the tests are skipped. Validation for SLES with preinstalled docker 19.03.x will be done separated.
